### PR TITLE
More warnings fixes

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperation.java [hunk 1]
```diff
@@ -51,18 +51,18 @@ protected IPath getModuleDeployDirectory(IModule module) {
...
-    return server.getPublishedResourceDelta(module);
+  protected IModuleResourceDelta[] getPublishedResourceDelta(IModule[] modules) {
+    return server.getPublishedResourceDelta(modules);
   }
 
   @Override
-  protected IModuleResource[] getResources(IModule[] module) {
-    return server.getResources(module);
+  protected IModuleResource[] getResources(IModule[] modules) {
+    return server.getResources(modules);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperation.java]: ditto

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperation.java [hunk 2]
```diff
@@ -51,18 +51,18 @@ protected IPath getModuleDeployDirectory(IModule module) {
...
-    server.setModulePublishState2(module, publishState);
+  protected void setModulePublishState(IModule[] modules, int publishState) {
+    server.setModulePublishState2(modules, publishState);
   }
 
   @Override
-  protected IModuleResourceDelta[] getPublishedResourceDelta(IModule[] module) {
-    return server.getPublishedResourceDelta(module);
+  protected IModuleResourceDelta[] getPublishedResourceDelta(IModule[] modules) {
+    return server.getPublishedResourceDelta(modules);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperation.java]: ditto

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperation.java [hunk 3]
```diff
@@ -51,18 +51,18 @@ protected IPath getModuleDeployDirectory(IModule module) {
   }
 
   @Override
-  protected void setModulePublishState(IModule[] module, int publishState) {
-    server.setModulePublishState2(module, publishState);
+  protected void setModulePublishState(IModule[] modules, int publishState) {
+    server.setModulePublishState2(modules, publishState);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperation.java]: In WTP, the array represents a module path to the module.  For example, module[0] = a web module, module[1] = utility module; the last module is the actual module.  So these should remain the singular `module`.

**Comment by elharo** [comment]
Not sure I follow re module or modules. If it should be singular we still
have the warning that a field is shadowed, so we'll need to fix that some
other way.

--
Elliotte Rusty Harold
on a mobile phone so please excuse weird autocorrects

**Comment by briandealwis** [comment]
I think `submodule` would work.

**Comment by elharo** [comment]
If the array represents the modulePath, then how about `modulePath` as the name?

**Comment by briandealwis** [comment]
Sounds good to me!

**Pull request merged**